fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,895 @@
+
+### 4. Examples
+
+Several scenarios will be covered in this sections showing how FSPs and a Mojaloop Switch would handle the generation and propagation of Trace headers depending on their respective participation.
+
+For all examples, the following _Node.js_ code was used to randomly generated unique _trace-id_ and _parent-id_ values:
+
+```javascript
+'use strict'
+
+const crypto = require('crypto')
+
+const Crypto = {
+  randomBytes (byteSize, encoding = 'hex') {
+    return crypto.randomBytes(byteSize).toString(encoding)
+  }
+}
+```
+
+Usage for _trace-id_ using default `hex` output encoding:
+
+```javascript
+console.log('trace-id='+Crypto.randomBytes(16))
+
+```
+
+Usage for _parent-id_ using default `hex` output encoding:
+```javascript
+console.log('parent-id='+Crypto.randomBytes(8))
+```
+
+#### 4.1. P2P Transfer with participating FSPs
+
+The following example will expand on the [FSPIOP API P2P Transfer](./API%20Definition%20v1.1.html#5161-p2p-transfer)<sup>[1](./API%20Definition%20v1.1.html#5161-p2p-transfer)</sup> by including distributed tracing.
+
+Each participant will be assigned a _vendor-name_ will be used throughout this example as per [Table 5](#table-5-–-p2p-transfer-participants).
+
+##### **Table 5 – P2P Transfer Participants**
+
+Participant | Identifier | _vendor-name_
+------|-----------|----------
+Payer FSP | BankNrOne | banknrone
+Payee FSP | MobileMoney | mobilemoney
+Mojaloop Payment Switch | Switch | moja
+
+
+##### 4.1.1. Request Party Information
+
+4.1.1.1. Request Party Information from BankNrOne to Switch
+
+BankNrOne sends a **GET /parties** request to the Switch, and includes the following generated _traceparent_ information:
+
+| variable | value | description |
+| ---|---|--- |
+| _version_ | `00` | Indicates Standard version as per [Table 3](#table-3-–-supported-version-formats) is being used. |
+| _trace-id_ | `91e502e28cd723686e9940bd3f378f85` | Unique trace identifier randomly generated. Refer to [Table 2](#table-2-–-data-model-for-tracing-values). |
+| _parent-id_ | `b0f903d000944947` | Updated to represent the current span. Refer to [Table 2](#table-2-–-data-model-for-tracing-values). |
+| _trace-flags_ | `01 `| Indicates that BankNrOne has recorded the trace. |
+
+The _tracestate_ is also include with the following information:
+| index | _list-member_ |value | description|
+| --- | --- | --- | --- |
+| 0 | `banknrone` | `b0f903d000944947` | Mapped from _parent-id_. |
+
+The following trace headers will be include in the request:
+
+**Note:** The fields highlighted below as **`value`** will indicate a newly generated or modified value depending on context. Non-highlighted fields will indicate an unchanged value.
+
+> traceparent: **`00`**-**`91e502e28cd723686e9940bd3f378f85`**-**`b0f903d000944947`**-**`01`**<br/>
+> tracestate: **`banknrone`**=**`b0f903d000944947`**<br/>
```

Comment:
nit: formatting 

```suggestion
- `traceparent`: **`00`**-**`91e502e28cd723686e9940bd3f378f85`**-**`b0f903d000944947`**-**`01`**<br/>
- `tracestate`: **`banknrone`**=**`b0f903d000944947`**<br/>

> ***Note:** The fields highlighted below as **`value`** will indicate a newly generated or modified value depending on context. Non-highlighted fields will indicate an unchanged value.*
```

Reply:
I am happy to make this change if the majority prefers it.

@elnyry-sam-k , @ehenrka. Your thoughts on this suggestion?

Reply:
I'm fine with the suggestion. Two additional comments:
- The "below" in "The fields highlighted below .." should probably be changed to "in the subsequent sections" or similar, as I assume this note is for all examples?
- The **`value`** in the note is in **bold**, but values in the tables seems to be in normal (non-bold) formatting. See for example the two tables in 4.1.1.2.